pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -0,0 +1,502 @@
+	templatesDir string
+
+	client        mcfgclientset.Interface
+	eventRecorder record.EventRecorder
+
+	syncHandler                   func(mcp string) error
+	enqueueContainerRuntimeConfig func(*mcfgv1.ContainerRuntimeConfig)
+
+	ccLister       mcfglistersv1.ControllerConfigLister
+	ccListerSynced cache.InformerSynced
+
+	mccrLister       mcfglistersv1.ContainerRuntimeConfigLister
+	mccrListerSynced cache.InformerSynced
+
+	mcpLister       mcfglistersv1.MachineConfigPoolLister
+	mcpListerSynced cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+}
+
+// New returns a new container runtime config controller
+func New(
+	templatesDir string,
+	mcpInformer mcfginformersv1.MachineConfigPoolInformer,
+	ccInformer mcfginformersv1.ControllerConfigInformer,
+	mcrInformer mcfginformersv1.ContainerRuntimeConfigInformer,
+	kubeClient clientset.Interface,
+	mcfgClient mcfgclientset.Interface,
+) *Controller {
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(&coreclientsetv1.EventSinkImpl{Interface: kubeClient.CoreV1().Events("")})
+
+	ctrl := &Controller{
+		templatesDir:  templatesDir,
+		client:        mcfgClient,
+		eventRecorder: eventBroadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: "machineconfigcontroller-containerruntimeconfigcontroller"}),
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineconfigcontroller-containerruntimeconfigcontroller"),
+	}
+
+	mcrInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    ctrl.addContainerRuntimeConfig,
+		UpdateFunc: ctrl.updateContainerRuntimeConfig,
+		DeleteFunc: ctrl.deleteContainerRuntimeConfig,
+	})
+
+	ctrl.syncHandler = ctrl.syncContainerRuntimeConfig
+	ctrl.enqueueContainerRuntimeConfig = ctrl.enqueue
+
+	ctrl.mcpLister = mcpInformer.Lister()
+	ctrl.mcpListerSynced = mcpInformer.Informer().HasSynced
+
+	ctrl.ccLister = ccInformer.Lister()
+	ctrl.ccListerSynced = ccInformer.Informer().HasSynced
+
+	ctrl.mccrLister = mcrInformer.Lister()
+	ctrl.mccrListerSynced = mcrInformer.Informer().HasSynced
+
+	return ctrl
+}
+
+// Run executes the container runtime config controller.
+func (ctrl *Controller) Run(workers int, stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer ctrl.queue.ShutDown()
+
+	glog.Info("Starting MachineConfigController-ContainerRuntimeConfigController")
+	defer glog.Info("Shutting down MachineConfigController-ContainerRuntimeConfigController")
+
+	if !cache.WaitForCacheSync(stopCh, ctrl.mcpListerSynced, ctrl.mccrListerSynced, ctrl.ccListerSynced) {
+		return
+	}
+
+	for i := 0; i < workers; i++ {
+		go wait.Until(ctrl.worker, time.Second, stopCh)
+	}
+
+	<-stopCh
+}
+
+func (ctrl *Controller) updateContainerRuntimeConfig(oldObj interface{}, newObj interface{}) {
+	oldKC := oldObj.(*mcfgv1.ContainerRuntimeConfig)
+	newKC := newObj.(*mcfgv1.ContainerRuntimeConfig)
+	if oldKC != newKC {
```

Comment:
all of this seems to still refers to `KC` or kubeconfig right?

Reply:
Yup, thanks! I missed it